### ISSUE TITLE
consents are not transferred to other newly registered users

### ISSUE THREAD

**Issue by KateChuen** [SAP/cloud-commerce-spartacus-storefront#5006]
**Environment Details**
Spartacus: epic/anonymous-consents branch
Browser: chrome
Device: mac
Happens on: http://dev-com-14.accdemo.b2c.ydev.hybris.com:4200/electronics-spa/en/USD/

**Steps to Reproduce**
- On homepage, scroll down and click on "Consent Preferences"
- Click on Accept All.
- Click on Sign/Register link and Register a new user.
- Login with that user. 
- Go to Consent Management Page -> All 3 consents are checked as expected.
- Go to homepage. Logout. 
- Click on "Consent Preferences" -> All 3 consents still checked as expected.
- Click on SignIn/Register link, and register another user (i did not check the consent on registration page).
- Login with that new user.
- Go to Consent Management Page 

**Actual Result**
- The consents are not checked.

**Expected Result**
- All 3 consents should be checked like for the first user.

**Comment by Xymmer** [comment]
hmmm i think the registration consent "not checked" takes precedence over what's in the anonymous. @znikola to confirm

**Comment by znikola** [comment]
That's correct. However, since all the anonymous consents are checked, it's expected that those are transferred to the second user as well. Will have to look into this one before the final release.

**Comment by Zeyber** [comment]
QA: Went to home page and accepted all consents
Logged in with user (all consents were checked)
Registered new user and checked consent preferences (all consents were checked)
